apps/aehttp/test/aehttp_client.erl

```diff
@@ -0,0 +1,100 @@
+-module(aehttp_client).
+
+-export([
+         request/3,
+         request/7
+        ]).
+
+%%=============================================================================
+%% API
+%%=============================================================================
+
+%% Required config values: int_addr, ext_addr
+%% Optional config values: encode_base58c
+request(OpId, Params, Cfg) ->
+    Op = endpoints:operation(OpId),
+    {Method, Interface} = operation_spec(Op),
+    BaseUrl = proplists:get_value(Interface, Cfg),
+    Params1 = maybe_encode_params(Params, Cfg),
+    Path = operation_path(Method, OpId, Params1),
+    request(Method, BaseUrl, Path, Params1, [], [], []).
+
+request(get, BaseUrl, Path, QueryParams, Headers, HttpOpts, Opts) ->
+    Url = make_url(BaseUrl, Path, QueryParams),
+    ct:log("GET ~p", [Url]),
+    Resp = httpc:request(get, {Url, Headers}, HttpOpts, Opts),
+    process_response(Resp);
+request(post, BaseUrl, Path, BodyParams, Headers, HttpOpts, Opts) ->
+    Url = make_url(BaseUrl, Path, #{}),
+    {ContentType, Body} = make_body(BodyParams, Path),
+    ct:log("POST ~p~nContentType ~p~n~p", [Url, ContentType, BodyParams]),
+    Resp = httpc:request(post, {Url, Headers, ContentType, Body}, HttpOpts, Opts),
+    process_response(Resp).
+
+%%=============================================================================
+%% Internal functions
+%%=============================================================================
+
+process_response({ok, {{_, Code, _State}, _Head, Body}}) ->
+    try
+        case iolist_to_binary(Body) of
+            <<>> ->
+                ct:log("Return code: ~p", [Code]),
+                {ok, Code, #{}};
+            B ->
+                B1 = jsx:decode(B, [return_maps]),
+                ct:log("Return code: ~p, body: ~p", [Code, B1]),
+                {ok, Code, B1}
+        end
+    catch
+        error:E -> {error, {parse_error, [E, erlang:get_stacktrace()]}}
+    end;
+process_response({error, _} = Error) ->
+    Error.
+
+maybe_encode_params(Params, Cfg) ->
+    case proplists:get_value(encode_base58c, Cfg) of
+        true -> encode_params(Params);
+        _ -> Params
+    end.
+
+%% TODO: add more params.
+encode_params(#{recipient_pubkey := PubKey} = Params) ->
+    EncPubKey = aec_base58c:encode(account_pubkey, PubKey),
+    maps:update(recipient_pubkey, EncPubKey, Params).
+
+operation_spec(#{get := GetSpec}) ->
+    {get, operation_interface(GetSpec)};
+operation_spec(#{post := PostSpec}) ->
+    {post, operation_interface(PostSpec)}.
+
+operation_interface(#{tags := Tags}) ->
+    IsExt = lists:member(<<"external">>, Tags),
+    IsInt = lists:member(<<"internal">>, Tags),
+    case {IsExt, IsInt} of
+        {true, _} -> ext_addr;
```

Comment:
I see that, as a client, if I want to invoke an HTTP API on an old version of the node I can rather use request/7 so looks ok. (I am thinking e.g. to bd8a10ba2b4d2102ad7ac9147352f220ef762d63 .)

Reply:
Or you can just pass `ext_addr` and `int_addr` of the old node in `Cfg` param of `request/3`.